### PR TITLE
grcqt: Minor Issues Fixes

### LINKED ISSUE #6957: feature-grc-qt Minor Issues
This is a collection of minor issues or notes found while testing the new Qt version of GRC.

Last checked on 4e444b7ff

- [x] Properties menu does not open when using a block's right click menu
- [x] Opening a GRC file immediately modifies it so it cannot be closed without saving
- [x] Right clicking on an unselected block does not select it and give access to options like enable, disable, properties

### PR DESCRIPTION
**Pull request by AsriFox**
## Description
Fixes for minor issues with the Qt version of GRC:
- [x] Ctrl+Scroll zooms in on the top left corner. Maybe it should zoom in on where the mouse is?
- [x] Zooming with Ctrl +, Ctrl -, Ctrl 0 is not implemented
- [x] Generate fails if the flowgraph hasn't been saved
- [x] Right clicking on an unselected block does not select it and give access to options like enable, disable, properties

Note: "Zoom in" shortcut is set to Ctrl =, because Ctrl + requires Shift to be pressed.

## Related Issue
Fixes #6957 

## Which blocks/areas does this affect?
GRC: Qt GUI

## Checklist
- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

File: grc/gui_qt/components/canvas/block.py [hunk 1]
```diff
@@ -443,6 +443,15 @@ def mousePressEvent(self, e):
...
         self.moveToTop()
 
+    def contextMenuEvent(self, e):
+        if not self.isSelected():
+            self.parent.clearSelection()
+            self.setSelected(True)
+
+        view = self.parent.view
+        contextMenu = view._app().MainWindow.menus["edit"]
+        pos = view.mapFromScene(e.pos())
```
Comment (haakov) [inline grc/gui_qt/components/canvas/block.py:453]: `e.pos()` is in block-relative coordinates, so you will have to map to scene coords before you can map to view coords

File: grc/gui_qt/components/window.py [hunk 2]
```diff
@@ -448,6 +445,25 @@ def createActions(self, actions):
         actions["properties"].setEnabled(False)
 
         # View Actions
+        actions["zoom_in"] = Action(
+            Icons("zoom-in"),
+            _("Zoom &in"),
+            self,
+            shortcut="Ctrl+=",  # TODO: Keys.ZoomIn is "Ctrl++"
```
Comment (haakov) [inline grc/gui_qt/components/window.py:452]: I think we should use Keys.ZoomIn, to stay consistent with GRC Gtk and other applications

**Review by haakov — CHANGES_REQUESTED** [review]
Thank you for the PR! 

I only have a few suggestions :smiley:

**Comment by AsriFox** [comment]
There are issues with GRC tests:
- GitHub actions fail because `pytest-qt` is not found in the container;
- some tests fail because of a [typo](https://github.com/gnuradio/gnuradio/blob/82c17682d4c0a95c0cecc94991aeb96d66dc5d23/grc/core/blocks/block.py#L62): there should be `parent_platform`, not `parent.platform`
- tests that involve `add_block_from_query` fail because the query does not produce a block; the text is typed in, but autocompletion doesn't work with qtbot's input (for some reason).

**Comment by haakov** [comment]
Great! Thank you for the typo, just fixed it in the feature-grc-qt branch. I'm moving the other issues you mentioned to https://github.com/gnuradio/gnuradio/issues/6957, so we still have it once this PR is closed.

**Comment by haakov** [comment]
https://github.com/gnuradio/gnuradio/blob/cfd798900fdd78beae8935ac304c083f0aaa256e/grc/gui_qt/components/flowgraph.py#L79

This should probably be set to False, since otherwise GRC will crash if you try to generate immedately after generating a new flowgraph

**Comment by haakov** [comment]
Merging once that's fixed

**Pull request merged**